homeassistant/components/energy/sensor.py

```diff
@@ -0,0 +1,131 @@
+"""Helper sensor for calculating utility costs."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Any
+
+from homeassistant.components.sensor import (
+    ATTR_LAST_RESET,
+    DEVICE_CLASS_MONETARY,
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+)
+from homeassistant.const import ATTR_NAME
+from homeassistant.core import HomeAssistant, callback, split_entity_id
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+import homeassistant.util.dt as dt_util
+
+from .data import async_get_manager
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the energy sensors."""
+    manager = await async_get_manager(hass)
+
+    if not manager.data:
+        return
+
+    entities = []
+
+    currency = manager.data["currency"]
+    for home_consumption in manager.data["home_consumption"]:
+        if home_consumption["stat_cost"]:
+            continue
```

Comment:
Instead of this, should we have a preference stating if the cost sensor should be generated automatically?

Reply:
This is ok for now.